### PR TITLE
refactor: Cleanup Sync generator

### PR DESCRIPTION
**Pull request by Youssef1313**
Purely refacotring PR.

## PR Type

What kind of change does this PR introduce?

- Refactoring (no functional changes, no api changes)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWPSyncGenerator/Generator.cs [hunk 1]
```diff
@@ -84,28 +150,22 @@ public virtual void Build(string basePath, string baseName, string sourceAssembl
...
-						  || Path.GetFileNameWithoutExtension(externalRedfs.Display).StartsWith("Microsoft.Web.WebView2.Core", StringComparison.Ordinal)
+						  let fileNameWithoutExtension = Path.GetFileNameWithoutExtension(externalRedfs.Display)
+						  where fileNameWithoutExtension.StartsWith("Windows.Foundation", StringComparison.Ordinal)
+						  || fileNameWithoutExtension.StartsWith("Microsoft.WinUI", StringComparison.Ordinal)
+						  || fileNameWithoutExtension.StartsWith("Microsoft.UI", StringComparison.Ordinal)
+						  || fileNameWithoutExtension.StartsWith("Microsoft.Foundation", StringComparison.Ordinal)
+						  || fileNameWithoutExtension.StartsWith("Microsoft.ApplicationModel.Resources", StringComparison.Ordinal)
+						  || fileNameWithoutExtension.StartsWith("Microsoft.Graphics", StringComparison.Ordinal)
+						  || fileNameWithoutExtension.StartsWith("Windows.Phone.PhoneContract", StringComparison.Ordinal)
+						  || fileNameWithoutExtension.StartsWith("Windows.Networking.Connectivity.WwanContract", StringComparison.Ordinal)
```
Comment (Youssef1313) [inline src/Uno.UWPSyncGenerator/Generator.cs:161]: I removed some redundant `StartsWith` checks here.

For example, we had previously `StartsWith("Microsoft.UI")`, `StartsWith("Microsoft.UI.Text")` checks. But the second is redundant. The first one is sufficient alone.

**Review by jeromelaban — APPROVED** [review]
Thanks!

File: src/Uno.UWPSyncGenerator/Generator.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UWPSyncGenerator/Generator.cs:161]: Looks good. It was probably an incorrect duplicate.

**Comment by Youssef1313** [comment]
@jeromelaban Can we merge this?

**Pull request merged**